backends/core/models.py

```diff
@@ -267,40 +373,159 @@ def get_ready(self):
           'message': 'Bad job param "%s": %s' % (param.label, e),
       })
       return False
-    self.update(status='waiting', status_changed_at=datetime.now())
-    return True
+    if self.prepare_for_start():  
+      return True
+    else:
+      logger.log_struct({
+          'labels': {
+              'pipeline_id': self.pipeline_id,
+              'job_id': self.id,
+              'worker_class': self.worker_class,
+          },
+          'log_level': 'ERROR',
+          'message': 'Memcache error - could not update the status of the job',
+      })
+      return False
+  
+  def _set_multi_cache(self, mapping, time=MEMCACHE_DEFAULT_EXPIRATION_TIME_SECONDS, max_retries=10):
```

Comment:
If you want to keep this name, then probably rename the other `_add_or_update_...` to match this one.

Btw, this class method looks like a state-less function to me because it's not depending on the Job instance values. Plus, you could reuse this function in the two model classes.

 So personally, I would define a function in a new file `backends/cache.py`:

```python
def set_multi_cache(mapping, key_prefix, time=..., max_retries=...):
   ...
```

This would simply the implementation here.

Reply:
You can also define a `def set_cache(...)` that will call this multi function.

Reply:
Concerning the memcache_client variable, I would suggest to put it as a shared instance.
For example in the `backends/cache.py` file:

```python

def get_memcache_client():
  return memcache.Client()

shared_memcache_client = get_memcache_client()
```

And then make use of this shared client in the `set_cache` and `set_multi_cache` functions.